scripts/test_mpc

```diff
@@ -4,42 +4,112 @@ set -x
+# Begin Phase2 MPC and run some rounds, accumulating transcript.
+${MPC} phase2-begin --out ${challenge_0_file} ${linear_combination_file}
+
+${MPC} phase2-contribute \
+       --skip-user-input \
+       --digest ${response_digest_1_file} \
+       --out ${response_1_file} ${challenge_0_file}
+${MPC} phase2-verify-contribution \
+       --transcript ${transcript_file} \
+       --new-challenge ${challenge_1_file} \
+       ${challenge_0_file} ${response_1_file}
+
+${MPC} phase2-contribute \
+       --skip-user-input \
+       --digest ${response_digest_2_file} \
+       --out ${response_2_file} ${challenge_1_file}
+${MPC} phase2-verify-contribution \
+       --transcript ${transcript_file} \
+       --new-challenge ${challenge_2_file} \
+       ${challenge_1_file} ${response_2_file}
+
+${MPC} phase2-contribute \
+       --skip-user-input \
+       --digest ${response_digest_3_file} \
+       --out ${response_3_file} ${challenge_2_file}
+${MPC} phase2-verify-contribution \
+       --transcript ${transcript_file} \
+       --new-challenge ${challenge_3_file} \
+       ${challenge_2_file} ${response_3_file}
+
+# Verify the transcript
+${MPC} phase2-verify-transcript \
+       ${challenge_0_file} ${transcript_file} ${challenge_3_file}
+
+# Verify the transcript, checking for each contribution
+${MPC} phase2-verify-transcript \
+       --digest ${response_digest_1_file} \
+       ${challenge_0_file} ${transcript_file} ${challenge_3_file}
+${MPC} phase2-verify-transcript \
+       --digest ${response_digest_2_file} \
+       ${challenge_0_file} ${transcript_file} ${challenge_3_file}
+${MPC} phase2-verify-transcript \
+       --digest ${response_digest_3_file} \
+       ${challenge_0_file} ${transcript_file} ${challenge_3_file}
+
+# Invalid contribution check
+echo 00000000 >> ${invalid_response_digest_file}
+cat ${response_digest_3_file} >> ${invalid_response_digest_file}
+if (${MPC} phase2-verify-transcript \
+       --digest ${invalid_response_digest_file} \
+       ${challenge_0_file} ${transcript_file} ${challenge_3_file}) then
+   echo ERROR: expected failure
+   exit 1
+fi
 
 # Create the keypair
-${MPC} \
-       create-keypair --out ${keypair_file} \
-       ${pot_file} ${linear_combination_file} ${phase2_file}
-
-rm \
-    ${pot_file} \
-    ${lagrange_file} \
-    ${linear_combination_file} \
-    ${phase2_file} \
-    ${keypair_file}
+${MPC} create-keypair --out ${keypair_file} \
+       ${pot_file} ${linear_combination_file} ${final_phase2_file}
+
+# Clean up files
+rm ${pot_file} ${lagrange_file} ${linear_combination_file}
+rm ${transcript_file}
+rm ${challenge_0_file}
+rm ${response_1_file} ${challenge_1_file} ${response_1_digest_file}
+rm ${response_2_file} ${challenge_2_file} ${response_2_digest_file}
+rm ${response_3_file} ${challenge_3_file} ${response_3_digest_file}
+invalid_response_digest_file=_test_response_digest_invalid.bin
+rm -f ${final_phase2_file}      # Same as challenge_3_file
```

Comment:
Let's remove the tabs/multiple spaces before the comment

Reply:
Removed spaces before comment